libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2016,49 +2013,64 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 				toString(parameterTypes.size()) +
 				".";
 
-		// Extend error message in case we try to construct a struct with mapping member.
-		if (isStructConstructorCall)
-		{
-			/// For error message: Struct members that were removed during conversion to memory.
-			TypePointer const expressionType = type(_functionCall.expression());
-			TypeType const& t = dynamic_cast<TypeType const&>(*expressionType);
-			auto const& structType = dynamic_cast<StructType const&>(*t.actualType());
-			set<string> membersRemovedForStructConstructor = structType.membersMissingInMemory();
+			// Extend error message in case we try to construct a struct with mapping member.
+			if (isStructConstructorCall)
+			{
+				/// For error message: Struct members that were removed during conversion to memory.
+				TypePointer const expressionType = type(_functionCall.expression());
+				TypeType const& t = dynamic_cast<TypeType const&>(*expressionType);
```

Comment:
Old code, but is good example of usage for auto:
```suggestion
				auto const& t = dynamic_cast<TypeType const&>(*expressionType);
```